modules/db/src/main/scala/gem/dao/StepDao.scala

```diff
@@ -171,16 +212,10 @@ object StepDao {
              LEFT OUTER JOIN step_smart_gcal ss
                 ON ss.step_smart_gcal_id = s.step_id
        WHERE s.observation_id = $oid
-    ORDER BY s.location
-    """.query[StepKernel].map(_.toStep).list
+    """.query[(Loc, StepKernel)].map(_.map(_.toStep)).list.map(==>>.fromList(_))
```

Comment:
Yes, it takes a list of tuples where the first value is used as the key (and so has to have an `Order` instance in scope).  In this case, the keys are `Location.Middle` so I had to define an `Order` instance for that even though `Location` itself is already ordered.